internal/webapi/webapi.go

```diff
@@ -97,23 +98,30 @@ func Start(ctx context.Context, requestShutdown func(), shutdownWg *sync.WaitGro
 	// use them to initialize the address generator.
 	idx, err := vdb.GetLastAddressIndex()
 	if err != nil {
-		return fmt.Errorf("db.GetLastAddressIndex error: %w", err)
+		return nil, fmt.Errorf("db.GetLastAddressIndex error: %w", err)
 	}
 	feeXPub, err := vdb.FeeXPub()
 	if err != nil {
-		return fmt.Errorf("db.GetFeeXPub error: %w", err)
+		return nil, fmt.Errorf("db.GetFeeXPub error: %w", err)
 	}
 	addrGen, err := newAddressGenerator(feeXPub, cfg.Network.Params, idx, log)
 	if err != nil {
-		return fmt.Errorf("failed to initialize fee address generator: %w", err)
+		return nil, fmt.Errorf("failed to initialize fee address generator: %w", err)
 	}
 
 	// Get the secret key used to initialize the cookie store.
 	cookieSecret, err := vdb.CookieSecret()
 	if err != nil {
-		return fmt.Errorf("db.GetCookieSecret error: %w", err)
+		return nil, fmt.Errorf("db.GetCookieSecret error: %w", err)
 	}
 
+	// Create TCP listener.
+	listener, err := net.Listen("tcp", cfg.Listen)
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("Listening on %s", cfg.Listen)
```

Comment:
Good idea. Implemented in a new commit but used `listener.Addr()` instead of `cfg.Listen`.

Reply:
> Good idea. Implemented in a new commit but used `listener.Addr()` instead of `cfg.Listen`.

Yeah not sure why I didn't put that in the example given it's right in the code I pasted above it 🥇